pandas/io/stata.py

```diff
@@ -2222,17 +2286,52 @@ def _write_bytes(self, value: bytes) -> None:
         """
         self.handles.handle.write(value)  # type: ignore[arg-type]
 
+    def _prepare_non_cat_value_labels(self, data: DataFrame) -> None:
```

Comment:
this is modifying global state and is pretty opaque to the reader. ideally these would return values and state would be changed in the caller.

Reply:
all of the state change to the caller.